File: padroesCriacionais/padrao-de-projeto-singleton/php/tests/EmailTest.php
<?php

namespace singleton\php;

use Singleton\Php\Email;
use Singleton\Php\Pessoa;
use PHPUnit\Framework\TestCase;

class EmailTest extends TestCase
{
    public function testDeveRetornarUmaInstanciaDeEmail(): void {
        $email = Email::getInstance();
        $this->assertInstanceOf(Email::class, $email);
    }

    public function testDeveRetornarMesmoObjetoEmailSempre(): void {
        $email1 = Email::getInstance();
        $email2 = Email::getInstance();
        $this->assertSame($email1, $email2);
    }

    public function testDeveLancarEmailComSucesso(): void {
        $email = Email::getInstance();
        $this->assertEquals(
            "...realiza a lógica de enviar os dados a partir de alguma biblioteca de envio de email",
            $email->send(
                "matricula",
                "matricula aprovada",
                (new Pessoa()),
                (new Pessoa())
            )
        );
    }
}

File: padroesCriacionais/padrao-de-projeto-singleton/java/src/test/java/EmailTest.java
import org.junit.jupiter.api.Test;
import singleton.Aluno;
import singleton.Email;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

public class EmailTest {
    @Test
    void deveRetornarUmaInstanciaDeEmail() {
        Email email = Email.getInstance();
        if(email == null) {
            fail();
        }
    }

    @Test
    void deveRetornarOMesmoObjetoEmailSempre() {
        Email email1 = Email.getInstance();
        Email email2 = Email.getInstance();
        if(email1 != email2) {
            fail();
        }
    }

    @Test
    void deveLancarUmEmailComSucesso() {
        Email email = Email.getInstance();
        assertEquals(
                "...realiza a lógica de enviar os dados a partir de alguma biblioteca de envio de email",
                email.send(
                        "matricula",
                        "matricula aprovada",
                        (new Aluno()),
                        (new Aluno())
                )
        );
    }
}
